app/controllers/ogi_pilot_sessions_controller.rb

```diff
@@ -3,96 +3,76 @@
 
 class OgiPilotSessionsController < ApplicationController
 
-  $publish_client = IsbmRestAdaptor::ProviderPublication.new
-  $subscribe_client = IsbmRestAdaptor::ConsumerPublication.new
-
   def new
-    @session = OgiPilotSession.new
-    logger.debug "NEW session initialized: #{@session.inspect}"
+    session = OgiPilotSession.new
+    logger.debug "NEW session initialized: #{session.inspect}"
+    render(SessionCreationComponent.new(session: session))
   end
 
   def create
-    @session = OgiPilotSession.new(ogi_pilot_session_params)
-    if @session.save
-      redirect_to ogi_pilot_sessions_path, notice: 'Session was successfully created.'
+    session  = OgiPilotSession.new(ogi_pilot_session_params)
+    if session.save
+      redirect_to ogi_pilot_sessions_path
     else
-      render :new, status: :unprocessable_entity
+      render turbo_stream: turbo_stream.replace(
+        'session_form',
+        renderable: SessionCreationComponent.new(session: session)
+      ), status: :unprocessable_entity
     end
   end
 
   def index
     @sessions = OgiPilotSession.all
-    Typhoeus::Config.verbose = true
-    IsbmRestAdaptor.configure do |conf|
-      conf.host = 'isbm.lab.oiiecosystem.net'
-      conf.scheme = 'https'
-      conf.base_path = '/rest'
-      # conf.debugging = true
-      conf.ssl_ca_cert = "C:/Program Files/curl/curl-ca-bundle.crt"
-    end
   end
 
   def edit
-    @session = OgiPilotSession.find(params[:id])
+    session = OgiPilotSession.find(params[:id])
+    render(SessionCreationComponent.new(session: session))
   end
 
   def update
-    @session = OgiPilotSession.find(params[:id])
+    session = OgiPilotSession.find(params[:id])
 
-    if @session.update(ogi_pilot_session_params)
-      redirect_to ogi_pilot_sessions_path, notice: 'Session was successfully updated.'
+    if session.update(ogi_pilot_session_params)
```

Comment:
You probably want to close the session (if it is open) before updating the object to avoid leaving invalid open sessions around. Be kind to the ISBM and release its resources ;-)
Could either do it here or in a before update hook on the OgiPilotSesssion class.